app/scripts/directives/raml-editor-save-file-button.js

```diff
@@ -21,20 +22,8 @@
           '</ul>'
         ].join('\n'),
         link:     function(scope) {
-          scope.contextMenuOpen = false;
-
           scope.openContextMenu = function () {
-            $timeout(function () {
-              $window.addEventListener('click', function self () {
-                scope.$apply(function () {
-                  scope.contextMenuOpen = false;
-                });
-
-                $window.removeEventListener('click', self);
-              });
-            });
-
-            scope.contextMenuOpen = true;
+            openSubMenuService.openSubMenu(scope, 'contextMenuOpen');
```

Comment:
Services handle entities, not verbs, the name of the service should not include a verb in it, the function name is fine.

I would call it just SubMenuService.open()